polyfill/lib/datetime.mjs

```diff
@@ -293,26 +293,31 @@ export class DateTime {
     );
   }
 
-  static fromString(isoString) {
-    isoString = ES.ToString(isoString);
-    const match = STRING.exec(isoString);
-    if (!match) throw new RangeError(`invalid datetime: ${isoString}`);
-    const year = ES.ToInteger(match[1]);
-    const month = ES.ToInteger(match[2]);
-    const day = ES.ToInteger(match[3]);
-    const hour = ES.ToInteger(match[4]);
-    const minute = ES.ToInteger(match[5]);
-    const second = ES.ToInteger(match[6]);
-    const millisecond = ES.ToInteger(match[7]);
-    const microsecond = ES.ToInteger(match[8]);
-    const nanosecond = ES.ToInteger(match[9]);
-    const DateTime = ES.GetIntrinsic('%Temporal.DateTime%');
-    const Construct = this;
-    return new Construct(year, month, day, hour, minute, second, millisecond, microsecond, nanosecond, 'reject');
-  }
-  static from(...args) {
-    const result = ES.CastDateTime(...args);
-    return this === DateTime ? result : new this(result.year, result.month, result.day);
+  static from(arg) {
+    if (typeof arg === 'object') {
+      const result = ES.CastDateTime(arg);
```

Comment:
I'm not a fan of having `CastDateTime` and `DateTime.from` be mutually recursive. Can we move all this into `CastDateTime`?